spec/services/request_password_reset_spec.rb

```diff
@@ -28,9 +28,10 @@
     context 'when the user is found and confirmed' do
       it 'sends password reset instructions' do
         user = create(:user)
-        email = user.email_addresses.first.email
+        email_address = user.email_addresses.first
+        email = email_address.email
 
-        allow(User).to receive(:find_with_email).with(email).and_return(user)
+        allow(EmailAddress).to receive(:find_with_email).with(email).and_return(email_address)
```

Comment:
why do we need to stub this?

Reply:
The line below:

```ruby
expect(user).to receive(:set_reset_password_token).and_return('asdf1234')
```

...allows for this assertion:

```ruby
        expect(UserMailer).to receive(:reset_password_instructions).
          with(email, token: 'asdf1234').
          and_return(mail)
```